src/storage/petastorm_storage_engine.py

```diff
@@ -0,0 +1,137 @@
+# coding=utf-8
+# Copyright 2018-2020 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+from src.spark.session import Session
+from src.catalog.models.df_metadata import DataFrameMetadata
+from petastorm.etl.dataset_metadata import materialize_dataset
+from src.storage.abstract_storage_engine import AbstractStorageEngine
+from src.configuration.configuration_manager import ConfigurationManager
+
+from petastorm.unischema import dict_to_spark_row
+from petastorm.predicates import in_set, in_lambda
+from petastorm import make_reader
+from typing import Iterator, Dict
+from pathlib import Path
+
+
+class PetastormStorageEngine(AbstractStorageEngine):
+
+    def __init__(self):
+        """
+        Maintain a long live spark session and context.
+        """
+        self._spark = Session()
+        self.spark_session = self._spark.get_session()
+        self.spark_context = self._spark.get_context()
+
+    def _spark_url(self, table: DataFrameMetadata) -> str:
+        """
+        Generate a spark/petastorm url given a table
+        """
+        eva_dir = ConfigurationManager().get_value("core", "location")
+        output_url = os.path.join(eva_dir, table.name)
+
+        return Path(output_url).resolve().as_uri()
+
+    def create(self, table: DataFrameMetadata):
+        """
+        Create an empty dataframe in petastorm.
+        """
+        empty_rdd = self.spark_context.emptyRDD()
+
+        with materialize_dataset(self.spark_session,
+                                 self._spark_url(table),
+                                 table.schema.petastorm_schema):
+
+            self.spark_session.createDataFrame(empty_rdd,
+                                               table.schema.pyspark_schema) \
+                .coalesce(1) \
+                .write \
+                .mode('overwrite') \
+                .parquet(self._spark_url(table))
+
+    def write_row(self, table: DataFrameMetadata, rows: []):
```

Comment:
We need to change this APIs name. write_batch makes move sense I think

Reply:
Only change the API name or change the implementation?
I would prefer keeping row as dict for now.

Reply:
We can return a batch object.

Reply:
Just a suggestion, we can just name it `write`. `rows` is anyway a parameter.

Reply:
Yeah